### DISCUSSION TITLE
!float: left" is back- why?

### DISCUSSION THREAD

**Discussion by hgfrey** [primefaces/community#2897] [PrimeFaces] [answered]
With the introduction of version 14.0.0 of PrimeFaces, "float: left" has been removed from the ui-g-X CSS classes. There was a discussion back then (1486) that ended in the fact that float-left was not responsible and should therefore remain that way. 

With version 14.0.5 the “float: left” is back again. Why? 

With version 14.0.0 we have added additional CSS classes so that the template in the show case is adhered to, according to which every element with a ui-g-X class must be surrounded by an element with a ui-g CSS class. This is now causing us problems and, as a first step, means that we have to remove the “ui-g” CSS class that we added back then.

Why this back and forth? Are we doing something wrong here?

Here is our markup.
![image](https://github.com/user-attachments/assets/bee3c5a2-a365-46e7-9951-bdcfa4258c05)

And that has been the result since version 14.0.5.
![image](https://github.com/user-attachments/assets/8ce678a8-527b-47f5-86c5-a82fd1502b57)

The turquoise lines on the left are the two buttons that were previously below the text.

**Comment by melloware** [comment, marked as answer]
It was fixed with this fix in 14.0.5: https://github.com/primefaces/primefaces/issues/12475

For years that was in `grid.css` and I removed it with https://github.com/primefaces/primefaces/pull/10701 but that was **NOT** the correct fix. So I put the float:left back as PrimeTek had originally intended in `grid.css`

**Reply by hgfrey** [reply]
Thanks for the feedback.

Now what's wrong with our markup and/or the CSS classes? I was now firmly under the assumption that a ui-g-X CSS class must always be embedded in an element with a ui-g CSS class.

**Reply by melloware** [reply]
That is how it is used by PF. You can view the HTML yourself look at the Responsive example here: https://www.primefaces.org/showcase/ui/input/manyCheckbox.xhtml


```xml
<div class="ui-g">
	<div class="ui-g-12 ui-md-4">
		<div class="ui-chkbox ui-widget">
			<div class="ui-helper-hidden-accessible">
				<input id="j_idt249:grid2:0" name="j_idt249:grid2" type="checkbox" value="Miami" data-p-hl="manychkbox" data-p-grouped="true"/>
			</div>
			<div class="ui-chkbox-box ui-widget ui-corner-all ui-state-default">
				<span class="ui-chkbox-icon ui-icon ui-icon-blank ui-c"></span>
			</div>
		</div>
		<label for="j_idt249:grid2:0">Miami</label>
	</div>
	<div class="ui-g-12 ui-md-4">
		<div class="ui-chkbox ui-widget">
			<div class="ui-helper-hidden-accessible">
				<input id="j_idt249:grid2:1" name="j_idt249:grid2" type="checkbox" value="London" data-p-hl="manychkbox" data-p-grouped="true"/>
			</div>
			<div class="ui-chkbox-box ui-widget ui-corner-all ui-state-default">
				<span class="ui-chkbox-icon ui-icon ui-icon-blank ui-c"></span>
			</div>
		</div>
		<label for="j_idt249:grid2:1">London</label>
	</div>
	<div class="ui-g-12 ui-md-4">
		<div class="ui-chkbox ui-widget">
			<div class="ui-helper-hidden-accessible">
				<input id="j_idt249:grid2:2" name="j_idt249:grid2" type="checkbox" value="Paris" data-p-hl="manychkbox" data-p-grouped="true"/>
			</div>
			<div class="ui-chkbox-box ui-widget ui-corner-all ui-state-default">
				<span class="ui-chkbox-icon ui-icon ui-icon-blank ui-c"></span>
			</div>
		</div>
		<label for="j_idt249:grid2:2">Paris</label>
	</div>
</div>

```

**Reply by hgfrey** [reply]
We don't have manyCheckbox elements. We have submit buttons that need to be displayed responsively. 

Other than that, our markup looks pretty much the same as the manyCheckbox example. But the layout is not as expected. If I remove the float: left or the surrounding ui-g everything looks good again.

**Reply by melloware** [reply]
yep i was just suggesting that is how PrimeFaces uses `ui-g` as an example of the markup.

**Reply by hgfrey** [reply]
The showCase shows here "https://www.primefaces.org/showcase/ui/panel/grid.xhtml?jfwid=00edf" how to use Grid CSS. But we have to remove the surrounding ui-g so that the layout fits, because adding "float: left" again destroys this layout.

**Reply by melloware** [reply]
You should switch to PrimeFlex from ui-g as that is the legacy Responsive...

**Reply by hgfrey** [reply]
That's interesting. For selectManyCheckbox of all things, the style is again overwritten with float: none. This can be seen in the current show case.

![image](https://github.com/user-attachments/assets/0d14b1fd-b1fe-42c7-ac6d-c2f3f1609973)

**Reply by melloware** [reply]
Yep that component overrides the float:left

**Reply by melloware** [reply]
Here is the code from PrimeFaces 8.0 and float:left has been there for over 7 years.

https://github.com/primefaces/primefaces/blob/cd4fbdf1d9d4ae054da19b9a84001d0c34d142eb/src/main/resources/META-INF/resources/primefaces/grid/grid.css#L33-L47

**Reply by hgfrey** [reply]
I found your argument from https://github.com/orgs/primefaces/discussions/1486 correct that float: left is not responsive. That's why we also rebuilt our ui-grid when we switched to 14.0.0. Now we'll just put everything back, since ui-g and float: left influence each other.

**Reply by melloware** [reply]
yep i wouldn't use ui-g at all i would use PrimeFlex.  grid.css is deprecated as I keep saying.

**Reply by melloware** [reply]
See: https://primeflex.org/gridsystem

**Reply by hgfrey** [reply]
Deprecated means that grid.css is thrown out. Is there a schedule for this?

**Reply by melloware** [reply]
No its not being removed we are leaving it in there for legacy but we provide a tool to convert called PrimeFaces CLI: https://github.com/primefaces/primefaces/tree/master/primefaces-cli

**Reply by hgfrey** [reply]
That sounds good. I'll take a look at the tool. Thank you.
